core/models/user.py

```diff
@@ -11,9 +11,16 @@ class UserType(models.TextChoices):
         STAKEHOLDER = "stakeholder", _("Stakeholder")
         TREASURER = "treasurer", _("Treasurer")
 
+    class AgencyRole(models.TextChoices):
+        AGENCY_INPUTTER = "agency_inputter", _("Agency inputter")
```

Comment:
I think you should add this 2 in the user type instead of agency. I think it will be easier in this way to validate, to maintain  and to understand the logic.